app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsViewModel.kt

```diff
@@ -0,0 +1,226 @@
+            eventUseCases.createEvent(state.agendaItem)
+                .onSuccess {
+                    channel.send(UiEventDetailsEvent.ShowInfoToast(it.countOfSkippedMessages.toString()))
+                    channel.send(UiEventDetailsEvent.OnBackClick)
+                }
+                .onFailure {
+                    channel.send(UiEventDetailsEvent.ShowErrorMessage(R.string.default_event_details_error))
+                }
+            channel.send(UiEventDetailsEvent.OnBackClick)
+        }
+    }
+
+    private fun deleteEvent() {
+        viewModelScope.launch {
+            eventUseCases.deleteEvent(state.agendaItem.id)
+            channel.send(UiEventDetailsEvent.OnBackClick)
+        }
+    }
+
+    private fun changeMode() {
+        updateState(state.copy(isEditable = !state.isEditable))
+    }
+
+    private fun addPhoto(uri: Uri) {
+        val photos = mutableListOf<AgendaPhoto>()
+        photos.addAll(state.agendaItem.photos)
+        photos.add(AgendaPhoto.LocalPhoto(uri.toString()))
+        updateEvent(event = state.agendaItem.copy(photos = photos))
+    }
+
+    private fun addAttendee(email: String) {
+        updateState(state.copy(isLoading = true))
+        viewModelScope.launch {
+            eventUseCases.checkAttendee(email)
+                .onSuccess { attendee ->
+                    if (attendee != null) {
+                        val attendees = mutableListOf<Attendee>().apply {
+                            addAll(state.agendaItem.attendees)
+                            add(
+                                Attendee(
+                                    email = attendee.email,
+                                    fullName = attendee.fullName,
+                                    userId = attendee.userId,
+                                    eventId = state.agendaItem.id,
+                                    isGoing = true,
+                                    remindAt = 0L
+                                )
+                            )
+                        }
+                        updateEvent(event = state.agendaItem.copy(attendees = attendees))
+                        updateState(state.copy(isLoading = false))
+                    } else {
+                        updateState(
+                            state.copy(
+                                isLoading = false,
+                                isErrorEmail = true,
+                                emailLabel = R.string.wrong_email
+                            )
+                        )
+                    }
+                }
+                .onFailure {
+                    updateState(
+                        state.copy(
+                            isLoading = false,
+                            isErrorEmail = true,
+                            emailLabel = R.string.wrong_email
+                        )
+                    )
+                }
+        }
+    }
+
+    private fun updateEvent(event: AgendaItem.Event) {
```

Comment:
Whats the use of this method?

Sometimes you use this, some other times you use updateState

Its confusing

Also, I would have thought that "updateEvent" actually updates the event on the db and api